### PR TITLE
feat(client&server): add annual reset for locations

### PR DESCRIPTION
**Pull request by yusufkandemir**
Implemented a new mutation named `resetRetailLocation` and utilized it in the client. It first backs up the given location by putting related tables in the DB into separate CSV files, e.g. `books.csv`, `copies.csv`, etc., and by copying the receipt PDF files. Afterwards, it cleans up the DB entries and receipt files, which are safe to delete given we backed up the system in a safe place.
We don't have a use for it, but technically, we can make it possible to back up the system anytime without deleting the data. We could also implement a restore backup system as the backup is created in a way that allows it to be restored.

### Other changes

Clarified the storage-related configuration and made it accept a relative path.

Previously, it "seemed" like it was allowing relative paths as stated in https://github.com/ilmercatinodellibro/ilmercatinodellibro/issues/43#issuecomment-2107202712, but that was actually because the import book command was always using the `tmp-files` directory resolved relative to the server root. Now, the command respects the variable, which now accepts relative paths.

Receipt files are now stored under the relevant retail location directory, instead of all receipt files under the same directory.

Improved the logging for commands. The commands are now included as part of the app module, instead of being a separate one. So, we will now be able to use injection.

### CODE REVIEW

File: packages/client/src/layouts/authenticated-layout.vue
Comment (yusufkandemir) [inline packages/client/src/layouts/authenticated-layout.vue:516]: I am not sure if we should reload the page directly or after displaying a notification for a few seconds. We might also want to do this differently. But, we definitely need to do something like this to reset the caches. Please let me know what you think.

Comment (SirAuron) [reply packages/client/src/layouts/authenticated-layout.vue:516]: I think that resetting the cache is the best thing we can do.
Trying to deal with updating it all seems like unnecessary complex to me for something that happens just once per year.
Thou have my blessing to proceed like this

Comment (yusufkandemir) [reply packages/client/src/layouts/authenticated-layout.vue:516]: Do you mean reset all the caches but do not refresh the page or display a notification?

Comment (SirAuron) [reply packages/client/src/layouts/authenticated-layout.vue:516]: Sorry, I've seen your reply only now.
I meant that I am more than fine with refreshing the page entirely, since this operation will happen once a year.

Comment (yusufkandemir) [reply packages/client/src/layouts/authenticated-layout.vue]: Done 👍

**Comment by SirAuron** [comment]
Took a first look at the code and I think it is good. However before merging we should wait for https://github.com/ilmercatinodellibro/ilmercatinodellibro/pull/90 to be completed and merged and then rebase this PR ontop of development.

**Comment by SirAuron** [comment]
Ok @yusufkandemir I think that you can now rebase this one. From what I can see, there should just be a small conflict inside the `.env.example` after the rebase I'll test this one.

**Review by SirAuron — CHANGES_REQUESTED** [review]
Tested this multiple times and I think we are looking good 👏

There is just a thing that we need to add on which I asked also asked for Callo's opinion, which is: when the reset happens, do we need to reset Schools and their courses?
We agree that the answer to that shall be yes, because between different years the schools can change, some can be closed and others opened, and their courses changes overtime.
So alongside deleting the other things we shall also delete schools (and their courses) where `school.provinceCode like 'replaceWithRetailLocationId'`.

After this is done we can merge the PR.

**Review by SirAuron — APPROVED** [review]

**Pull request merged**
